scenariolib/event_pageview.go

```diff
@@ -46,6 +47,16 @@ func newViewEvent(e *JSONEvent) (*ViewEvent, error) {
 	}
 	event.clickRank = int(docNo)
 
+	if e.Arguments["pageViewField"] != nil {
+		if event.pageViewField, validcast = e.Arguments["pageViewField"].(string); !validcast {
+			return nil, errors.New("Parameter pageViewField must be of type string in ViewEvent")
+		}
+	} else if c.DefaultPageViewField != "" {
```

Comment:
Check also the method in `config.go` called `fillDefaults(c *Config)`
I follow the scheme that you just assign what values are present in the config received, and then fill in default values if some are needed but not provided.